### ISSUE TITLE
TypeError: d.getErrors is not a function: Build broken with 2.2.0-rc.4, but working with 2.1.0-beta.28

### ISSUE THREAD

**Issue by macmillan78** [webpack/webpack#3953]
**Do you want to request a *feature* or report a *bug*?**
BUG

**What is the current behavior?**
Get following stacktrace:
./node_modules/.bin/webpack --config ./config/webpack.config.production.js 
keywords if/then/else require v5 option
/path/node_modules/webpack/lib/Compilation.js:640
                                const errors = d.getErrors();
                                                 ^

TypeError: d.getErrors is not a function
    at block.dependencies.forEach.d (/path/node_modules/webpack/lib/Compilation.js:640:22)
    at Array.forEach (native)
    at blocks.forEach.block (/path/node_modules/webpack/lib/Compilation.js:632:23)
    at Array.forEach (native)
    at Compilation.reportDependencyErrorsAndWarnings (/path/node_modules/webpack/lib/Compilation.js:631:10)
    at modules.forEach.m (/path/node_modules/webpack/lib/Compilation.js:502:34)
    at Array.forEach (native)
    at Compilation.finish (/path/node_modules/webpack/lib/Compilation.js:502:16)
    at /path/node_modules/webpack/lib/Compiler.js:469:16
    at /path/node_modules/tapable/lib/Tapable.js:225:11
    at _addModuleChain (/path/node_modules/webpack/lib/Compilation.js:452:11)
    at processModuleDependencies.err (/path/node_modules/webpack/lib/Compilation.js:423:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)

d is:
ConstDependency {
  "module": null,
  "expression": "\"production\"",
  "range": [
    32190,
    32210
  ],
  "loc": SourceLocation {
    "start": {
      "line": 1119,
      "column": 22
    },
    "end": {
      "line": 1119,
      "column": 42
    }
  }
}

**If the current behavior is a bug, please provide the steps to reproduce.**
webpack config, just to give an overview: https://gist.github.com/macmillan78/6ab1411b5623d4fbf8106d9146060900

Seems to occur at every ConstDependency.

Changing
Compilation.js:640 from `const errors = d.getErrors()` to `const errors = d.getErrors && d.getErrors()` fixed the bug and produced a working build.

**What is the expected behavior?**
Build should work as it did with v2.1.0-beta.28

**Please mention other relevant information such as the browser version, Node.js version, Operating System and programming language.**
Node: 7.0.0
OS: macOS Sierra
webpack: v2.2.0-rc.4

**Comment by yhjor1212** [comment]
same here

**Comment by jure** [comment]
Fixed in latest https://github.com/webpack/webpack/releases/tag/v2.2.0-rc.7. Not sure why, I looked through the commits and it's not clear to me what fixed it.

**Comment by macmillan78** [comment]
Didn't fix it for me! As before, it works with the dev server, but it does not for a normal build using webpack CLI.

**Comment by jure** [comment]
Hmmm, interesting. My webpack build now isn't broken even in rc.4. Not sure what happened there. Looking into it, will report back.

**Comment by sokra** [comment]
Try nuking node_modules and reinstall

**Comment by jure** [comment]
If you're replying to me, I've tried everything but I can't reproduce this bug anymore with any of the rc.4, rc.5 and rc.6 family (and I could reliably reproduce it before upgrading to rc.7, it's so weird). Don't think anything change in my project that would affect this, but I'll recheck.

**Comment by jure** [comment]
**TLDR: Make sure your globally installed webpack version and the version your app depends on are both the latest, or at least match.**

Ok, I think this is a non issue. It happens when you're running new webpack code (say you have globally installed webpack 2.2.0-rc.5) and you're depending on old webpack versions (say 2.2.0-beta.28) in your app, and using that older version of webpack with say, DefinePlugin, e.g.:

```
new webpack.DefinePlugin({
  'process.env.NODE_ENV': JSON.stringify('production')
})
```

Because you `require('webpack')` in your config to get the DefinePlugin, you require the older version of webpack and the implementation of ConstDependency/Dependency that's missing getErrors.

The new `Compilation.js` depends on `Dependency.js`' `getErrors()` (https://github.com/webpack/webpack/blob/d9ac04380e04852b8e61592c431411663a0c658a/lib/Compilation.js#L640-L646), which was introduced in: https://github.com/webpack/webpack/blob/f745f02910cb45a55ed659ebd99288e45a38e873/lib/Dependency.js#L35-L37

Older versions of `Compilation.js` do not depend on `getErrors()` and only use `getWarnings()` (https://github.com/webpack/webpack/blob/92a8be23614f92bc5669bf43eb0ddf746b2477e7/lib/Compilation.js#L644-L650)

**Comment by macmillan78** [comment]
Updated EVERYTHING to final 2.2.0 and now it works!

Thx for the help!
